fix: produce JSONPatch patches with removals in the right order

fiam opened the pull request:
- Replace github.com/mattbaird/jsonpatch with github.com/wI2L/jsondiff
- Add support for live queries in the TS client
- Make subscriptions cancellable
- Enable jsonpatch for subscriptions over fetch
- Add integration test using jsonpatch over a live query

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md
Squashed commit must follow https://www.conventionalcommits.org/en/v1.0.0/ so we can generate the changelog.

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

## Motivation and Context

This caused a problem every time a live query deleted multiple entries at the end
of the list at the same time.

<!--
Why is this change required? What problem does it solve? Which issues are linked?
Please try to describe in detail the impact of this change. Add screenshots if it helps.
-->

## Marketing Changelog

Live Queries now work with bulk deletions

<!--
What changes does this PR introduce? Please describe the changes in simple terms that a user can understand
without being familiar with the codebase. Mention @advocates for review and tracking.
-->

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

jensneuse commented:
Would it not make sense to fork the upstream, provide a fix, and use our fork until it's merged?

fiam commented:
> Would it not make sense to fork the upstream, provide a fix, and use our fork until it's merged? I think someone mentioned it already: [mattbaird/jsonpatch#30](https://github.com/mattbaird/jsonpatch/issues/30)
> 
> So we can open a PR and help them improve their lib.

As discussed, we've added integration tests and replaced the library with another one.

StarpTech commented on packages/sdk/src/client/client.ts:
```diff
@@ -512,7 +516,18 @@ export class Client {
+				value = result.value;
+				done = result.done;
+			} catch (e) {
+				if (!(e instanceof Error) || !e.message.indexOf('aborted')) {
```
Could you add a comment for which case this is required? AbortController?

StarpTech reviewed (approved):
LGTM
